### DISCUSSION TITLE
Libraries required to parse SVG files

### DISCUSSION THREAD

**Discussion by D10f** [ImageMagick/ImageMagick#6831] [Help] [answered]
### ImageMagick version

6.9.11-60

### Operating system, version and so on

Debian 12

### Description

Hi!

I'd like to run a web app that leverages ImageMagick to do some basic image manipulation. I noticed though that some SVG files are too complex for ImageMagick, and I need to install Inkscape in order to get things working. The issue is that this increases my Docker image size considerably and I'd like to know which dependencies or libraries exactly would I need to have installed in order to have IM working properly, at the lowest disk impact as possible. Is this even possible or there's no other option than to install inkscape entirely?

My use case is actually quite simple this is the type of command that I'm running: `convert -background none input.svg -resize 32x32 -format png output.png`. Sometimes I would convert to webp or at different sizes as well, but for now this is all I really need.

Thank you!

**Comment by fmw42** [comment, marked as answer]
The issue is the complexity of the SVG file and features it uses.  Inkscape is the most versatile and best. Next is the RSVG delegate library. But it may not handle all the case that Inkscape does.  Lastly is nothing. IM has built-in the MSVG/XML method. But it is least accurate.  You would have to try each to find out if it handles your use case.

**Comment by D10f** [comment]
I see, well, I guess I'll stick with Inkscape for good measure and eventually try to narrow the problem down. The thing is I don't know in advance what the SVGs would look like so I do want to have the broadest possible support anyway. Thanks for clearing that up!
